src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewColumnCollectionDialog.cs

```diff
@@ -0,0 +1,1505 @@
+    ///  IComponent because some editors for some dataGridViewColumn properties - DataGridViewComboBox::DataSource editor -
+    ///  need the site
+    /// </summary>
+    internal class ListBoxItem : ICustomTypeDescriptor, IComponent
+    {
+        private DataGridViewColumn _column;
+        private DataGridViewColumnCollectionDialog _owner;
+        private ComponentDesigner? _compDesigner;
+        private Image? _toolboxBitmap;
+        public ListBoxItem(DataGridViewColumn column, DataGridViewColumnCollectionDialog owner, ComponentDesigner compDesigner)
+        {
+            _column = column;
+            _owner = owner;
+            _compDesigner = compDesigner;
+
+            if (_compDesigner is not null)
+            {
+                _compDesigner.Initialize(column);
+                TypeDescriptor.CreateAssociation(_column, _compDesigner);
+            }
+
+#pragma warning disable IL2077 // Target parameter argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The source field does not have matching annotations.
+            ToolboxBitmapAttribute? attr = TypeDescriptor.GetAttributes(column!)[_toolboxBitmapAttributeType!] as ToolboxBitmapAttribute;
+#pragma warning restore IL2077 // Target parameter argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The source field does not have matching annotations.
+            if (attr is not null)
+            {
+                _toolboxBitmap = attr.GetImage(column, false /*large*/);
+            }
+            else
+            {
+                _toolboxBitmap = SelectedColumnsItemBitmap;
+            }
+
+            DataGridViewColumnDesigner? dgvColumnDesigner = compDesigner as DataGridViewColumnDesigner;
+            if (dgvColumnDesigner is not null && _owner._liveDataGridView is not null)
+            {
+                dgvColumnDesigner.LiveDataGridView = _owner._liveDataGridView;
+            }
+        }
+
+        public DataGridViewColumn DataGridViewColumn
+        {
+            get
+            {
+                return _column;
+            }
+        }
+
+        public ComponentDesigner DataGridViewColumnDesigner
+        {
+            get
+            {
+                return _compDesigner!;
+            }
+        }
+
+        public DataGridViewColumnCollectionDialog Owner
+        {
+            get
+            {
+                return _owner;
+            }
+        }
+
+        public Image ToolboxBitmap
+        {
+            get
+            {
+                return _toolboxBitmap!;
+            }
+        }
+
+        public override string ToString()
+        {
+            return _column.HeaderText;
+        }
+
+        // ICustomTypeDescriptor implementation
+        AttributeCollection ICustomTypeDescriptor.GetAttributes()
+        {
+            return TypeDescriptor.GetAttributes(_column);
+        }
```

Comment:
```suggestion
        AttributeCollection ICustomTypeDescriptor.GetAttributes() => TypeDescriptor.GetAttributes(_column);
```
Ditto below as well.